packages/dds/tree/src/test/feature-libraries/modular-schema/modularChangeFamily.spec.ts

```diff
@@ -1048,6 +1064,170 @@ describe("ModularChangeFamily", () => {
 		});
 	});
 
+	describe("adds missing builds", () => {
+		const aMajor = mintRevisionTag();
+		const a1 = { major: aMajor, minor: 1 };
+		const a2 = { major: aMajor, minor: 2 };
+		const bMajor = mintRevisionTag();
+		const b1 = { major: bMajor, minor: 1 };
+		const cMajor = mintRevisionTag();
+		const c1 = { major: cMajor, minor: 1 };
+		const u1 = { minor: 1 };
+
+		const node2 = singleJsonCursor(2);
+		const node2Chunk = treeChunkFromCursor(node2);
+		const node3 = singleJsonCursor(3);
+		const node3Chunk = treeChunkFromCursor(node3);
+
+		const nodesArray: [DeltaDetachedNodeId, TreeChunk][] = [
+			[a1, node1Chunk],
+			[a2, node2Chunk],
+			[b1, node3Chunk],
+		];
+		const nodeMap = nestedMapFromFlatList(
+			nodesArray.map(([{ major, minor }, chunk]) => [major, minor, chunk]),
+		);
+
+		const getDetachedNode = ({ major, minor }: DeltaDetachedNodeId) => {
+			return tryGetFromNestedMap(nodeMap, major, minor);
+		};
+
+		it("with no existing builds", () => {
```

Comment:
The current factoring of the tests makes it hard to understand at a glance what the contract of the function is. It also makes it harder to verify that all relevant cases are being tested (and makes it more likely that coverage might lapse).

Here's a factoring that I think helps:
* it("preserves relevant builds that are present in the input")
* it("preserves irrelevant builds that are present in the input")
* describe("attemps to add relevant builds that are missing from the input")
  * it("adds the missing build if the detached node is available")
  * it("throws if the detached node is not available")

Reply:
Same idea for `filterSuperfluousBuilds`